### PR TITLE
Simulate: Support newer simulate options

### LINKED ISSUE #519: Support latest simulate features
Support the latest simulate features, similar to this JS PR: https://github.com/algorand/js-algorand-sdk/pull/818

### LINKED ISSUE #536: Missing allow_unnamed_resources in simulate
## Problem

We need to be able to call simulate without knowing the box references like the other sdks.

## Solution

Add the allow_unnamed_resources flag into the SimulateRequest method.

## Dependencies

I don't believe so.

## Urgency

The urgency is pretty high because the sdk is unusable for the majority of readonly method calls on smart contracts that have unknown box requirements. It is also very important for any method call when you don't know the box resources, because you need to do a simulate call first to get them.

### PR DESCRIPTION
**Pull request by jasonpaulos**
This SDK is behind on support for simulate options. This PR adds all simulate request options that are currently available in algod.

Similar to this JS PR: https://github.com/algorand/js-algorand-sdk/pull/818

Closes #519, closes #536

### CODE REVIEW

File: tests/steps/application_v2_steps.py [hunk 1]
```diff
@@ -24,6 +24,8 @@ def operation_string_to_enum(operation):
         return transaction.OnComplete.NoOpOC
     elif operation == "create":
         return transaction.OnComplete.NoOpOC
+    elif operation == "create-and-optin":
```
Comment (jannotti) [inline tests/steps/application_v2_steps.py:27]: How did this come to be called "create-and-optin"?

File: tests/steps/other_v2_steps.py [hunk 2]
```diff
@@ -1636,6 +1642,220 @@ def compare_avm_value_with_string_literal(string_literal, avm_value):
...
+        assert len(txn_group_path_split) > 0
+
+        traces = simulation_response["txn-groups"][0]["txn-results"][
+            txn_group_path_split[0]
+        ]["exec-trace"]
+        assert traces
+
+        for i in range(1, len(txn_group_path_split)):
+            traces = traces["inner-trace"][txn_group_path_split[i]]
+            assert traces
```
Comment (jannotti) [inline tests/steps/other_v2_steps.py:1771]: `for p in txn_group_path_split[1:]:` ?

File: tests/steps/other_v2_steps.py [hunk 3]
```diff
@@ -1636,6 +1642,220 @@ def compare_avm_value_with_string_literal(string_literal, avm_value):
...
+        int(p) for p in txn_group_path.split(",") if p != ""
+    ]
+    assert len(txn_group_path_split) > 0
+
+    traces = simulation_response["txn-groups"][0]["txn-results"][
+        txn_group_path_split[0]
+    ]["exec-trace"]
+    assert traces
+
+    for i in range(1, len(txn_group_path_split)):
```
Comment (jannotti) [inline tests/steps/other_v2_steps.py:1842]: Simpler iteration?

**Review by jannotti — APPROVED** [review]

File: tests/steps/application_v2_steps.py (hunk 1 above)
Comment (jasonpaulos) [reply tests/steps/application_v2_steps.py:27]: I added it when adding this cucumber test: https://github.com/algorand/algorand-sdk-testing/blob/7f06a6bd5fa352f87d62e108e2dc35bf05021343/features/integration/simulate.feature#L212

For legacy reasons, this step has "operations" which are the standard on completion options, and a "create" operation, which means noop but also don't include any app ID. I needed an app create that also opted in, so I chose to define another "operation" type that did that. Not the best solution but it was the easiest

File: tests/steps/other_v2_steps.py (hunk 3 above)
Comment (jasonpaulos) [reply tests/steps/other_v2_steps.py]: 9a7f9e4

File: tests/steps/other_v2_steps.py (hunk 2 above)
Comment (jasonpaulos) [reply tests/steps/other_v2_steps.py]: 9a7f9e4

**Pull request merged**